src/plugins/util/parser.ts

```diff
@@ -0,0 +1,159 @@
+import {
+	ArrayExpression,
+	AssignmentExpression,
+	BaseFunction,
+	Expression,
+	Property,
+	Statement,
+	SwitchCase,
+	VariableDeclaration
+} from 'estree';
+
+/**
+ * @private
+ * If the provided node is an array expression, then return an array containing its values.
+ *
+ * @param item
+ * An AST node
+ *
+ * @return
+ * An array of values if the node is an array expression; otherwise, `undefined`.
+ */
+export function extractArrayValues(item: any) {
+	if (!isArrayExpression(item)) {
+		return;
+	}
+	return item.elements.map((element: any) => element.value);
+}
+
+/**
+ * Return a parent node's child node, if it exists.
+ *
+ * @param item
+ * The parent node.
+ *
+ * @return
+ * The child node, if it exists.
+ */
+export const getNextItem = (function () {
+	const getters = {
+		argument: (item: { argument: Expression | null }) => item.argument,
+		arguments: (item: { arguments: Expression[] }) => item.arguments,
+		assignment: (item: AssignmentExpression) => item.right,
+		body: (item: { body: any }) => item.body,
+		call: (item: { callee: any, arguments: any[] }) => [ item.callee, ...(item.arguments || []) ],
+		'case': (item: SwitchCase) => item.consequent,
+		cases: (item: { cases: SwitchCase[] }) => item.cases,
+		conditional: (item: { alternate: Expression; consequent: Expression; }) => [ item.alternate, item.consequent ],
+		elements: (item: { elements: Expression[] }) => item.elements,
+		expression: (item: { expression: Expression }) => item.expression,
+		expressions: (item: { expressions: Expression[] }) => item.expressions,
+		loop: (item: { test: Expression; body: Statement; }) => [ item.test, item.body ],
+		object: (item: { object: Expression | null }) => item.object,
+		properties: (item: { properties: Property[] }) => item.properties,
+		value: (item: { value: Expression }) => item.value
+	};
+
+	const nextItemMap: { [key: string]: (item: any) => any; } = {
+		ArrayExpression: getters.elements,
+		AssignmentExpression: getters.assignment,
+		CallExpression: getters.call,
+		ConditionalExpression: getters.conditional,
+		DoWhileStatement: getters.loop,
+		ExpressionStatement: getters.expression,
+		IfStatement: getters.conditional,
+		MemberExpression: getters.object,
+		NewExpression: getters.arguments,
+		ObjectExpression: getters.properties,
+		Property: getters.value,
+		ReturnStatement: getters.argument,
+		SequenceExpression: getters.expressions,
+		SwitchCase: getters[ 'case' ],
+		SwitchStatement: getters.cases,
+		VariableDeclaration: (item: VariableDeclaration) => item.declarations,
+		WhileStatement: getters.loop
+	};
+
+	[
+		'ArrowExpression',
+		'ArrowFunctionExpression',
+		'BlockStatement',
+		'ForInStatement',
+		'ForOfStatement',
+		'ForStatement',
+		'FunctionDeclaration',
+		'FunctionExpression',
+		'LetStatement',
+		'Program',
+		'TryStatement'
+
+	].forEach((type: string) => {
+		nextItemMap[ type ] = getters.body;
+	});
+
+	return function (item: any): any {
```

Comment:
Item shouldn't be undefined unless something is wrong with the AST tree, but I added a check regardless.